core/src/consensus.rs

```diff
@@ -165,24 +164,47 @@ impl Tower {
             .map(|lockout| (lockout.stake as f64 / total_staked as f64) > self.threshold_size)
             .unwrap_or(false)
     }
-
-    pub fn record_vote(&mut self, slot: u64, hash: Hash) -> Option<u64> {
+    fn new_vote(
+        local_vote_state: &VoteState,
+        slot: u64,
+        hash: Hash,
+        bank_vote_state: &VoteState,
+    ) -> Vote {
+        let mut local_vote_state = local_vote_state.clone();
+        let vote = Vote {
+            slots: vec![slot],
+            hash,
+        };
+        local_vote_state.process_vote_unchecked(&vote);
+        let bank_slots = local_vote_state
```

Comment:
Is `local_vote_state.votes.len().max(bank_vote_state.votes.len())` better than an iter + zip? defer to you judgement.